internal/codegen/golang/opts/override.go

```diff
@@ -1,13 +1,79 @@
-package config
+package opts
 
 import (
 	"fmt"
 	"os"
 	"strings"
 
 	"github.com/sqlc-dev/sqlc/internal/pattern"
+	"github.com/sqlc-dev/sqlc/internal/plugin"
 )
 
+type PluginGoType struct {
+	ImportPath string
+	Package    string
+	TypeName   string
+	BasicType  bool
+	StructTags map[string]string
+}
+
+func pluginGoType(o *Override) *PluginGoType {
+	// Note that there is a slight mismatch between this and the
+	// proto api. The GoType on the override is the unparsed type,
+	// which could be a qualified path or an object, as per
+	// https://docs.sqlc.dev/en/v1.18.0/reference/config.html#type-overriding
+	return &PluginGoType{
+		ImportPath: o.GoImportPath,
+		Package:    o.GoPackage,
+		TypeName:   o.GoTypeName,
+		BasicType:  o.GoBasicType,
+		StructTags: o.GoStructTags,
+	}
+}
+
+type PluginOverride struct {
```

Comment:
I would rename this and the `PluginGoType` based on whatever you think a good name for the `Override` field is.

Reply:
I'm going to move both of these into shim.go